homeassistant/components/honeywell/__init__.py

```diff
@@ -30,14 +30,14 @@ async def async_setup_entry(hass, config):
     loc_id = config.data.get(CONF_LOC_ID)
     dev_id = config.data.get(CONF_DEV_ID)
 
-    devices = []
+    devices = {}
 
     for location in client.locations_by_id.values():
         for device in location.devices_by_id.values():
             if (not loc_id or location.locationid == loc_id) and (
```

Comment:
The locationid check can happen in the first loop to avoid entering the second loop fwiw.

Reply:
Done.